### PR TITLE
Fix uninitialized pointer in wxJoystick.

### PR DESCRIPTION
**Pull request by StepanHrbek**

### CODE REVIEW

File: src/msw/joystick.cpp [hunk 1]
```diff
@@ -157,6 +157,7 @@ wxJoystick::wxJoystick(int joystick)
     JOYINFO joyInfo;
     int i, maxsticks;
 
+    m_thread = nullptr;
```
Comment (PBfordev) [inline src/msw/joystick.cpp]:
```suggestion
    m_thread = NULL;
```

`nullptr` is C++11, wxWidgets is still C++98, so `NULL` must be used.

**Comment by PBfordev** [comment]
Sorry for not noticing it before but I wonder whether the added line would be better suited to be placed after `m_joystick = 0;` at the end of the ctor.

**Comment by StepanHrbek** [comment]
PBfordev, Thanks for nullptr explanation, I should have noticed it myself.

Regarding place for clearing pointers, both are good and I don't mind having it at the end in this simple case. But I think it's good practice to clear pointers and other simple member variables at the beginning. If you clear at the end, it's matter of time before future contributor adds exception or return from the middle of ctor and it's broken again.

**Comment by PBfordev** [comment]
I share the belief that if the variables cannot be initialized in their declarations, they should be initialized as closely to their declaration as possible. TBH, I would not mind if all member variables with cheap initialization were initialized before the ctor body.

OTOH, I really doubt exceptions would be introduced like that in wxWidgets and if there were a new  exit point added before the place I suggested, it would leave `m_joystick` uninitialized as well.

Well, it's just nitpicking at this point, sorry.

**Comment by vadz** [comment]
Thanks for noticing and fixing this!

**Pull request closed without merging**